doc/manual/src/design/store/store.md

```diff
@@ -1,10 +1,14 @@
-# Nix Store
+# Store
 
-A Nix store is a collection of *store objects* referred to by *store paths*.
-Every store also has a "store directory path", which is a path prefix used for various purposes.
+A Nix store is a collection of [store objects](objects.md) and associated operations.
 
-There are many types of stores, but all of them at least respect this model.
```

Comment:
I would like to specify that at least a little bit, but would not even know a different type of store.

Reply:
Start with
```
nix --extra-experimental-features nix-command describe-stores --json | jq keys
[
  "Dummy Store",
  "Http Binary Cache Store",
  "Legacy SSH Store",
  "Local Binary Cache Store",
  "Local Daemon Store",
  "S3 Binary Cache Store",
  "SSH Store"
]
```
Without `--json` it does markdown --- that markdown was intended to go in the manual along with some of the other auto-generated sections, but that has not happened yet.

Reply:
Concretely, `cache.nixos.org` is a "Http binary cache store", and remote builders are "SSH Store"s or "Legacy SSH Store"s. None of these expose to the client (the client itself implements the store interface) the file system, and `cache.nixos.org` in using S3 in fact may not use it behind the scenes either.

Reply:
Thanks, that's very helpful.